sklearn/tree/_tree.pyx

```diff
@@ -197,6 +197,16 @@ cdef class Criterion:
 
         pass
 
+    cdef double approx_impurity_improvement(self) nogil:
+        """Approximate the impurity reduction removing constants for a split"""
```

Comment:
+1, the method name is misleading. 

This thing is in fact a proxy quantity such that the split that maximizes this value also maximizes the impurity improvement.